### PR TITLE
Add preference to mute call ringtone

### PR DESCRIPTION
**Pull request by LuKks**
fix https://github.com/sindresorhus/caprine/issues/209

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#209: Call sound should also be disabled](https://issuehunt.io/repos/42574339/issues/209)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: source/index.ts [hunk 1]
```diff
@@ -233,6 +233,18 @@ function initRequestsFiltering(): void {
...
+
+	session.defaultSession.webRequest.onHeadersReceived({
+		urls: ['*://static.xx.fbcdn.net/rsrc.php/*']
+	}, ({responseHeaders}, callback) => {
+		if (!config.get('callRingtoneMuted') || !responseHeaders) {
+			return callback({});
+		}
+
+		callback({
+			cancel: responseHeaders['content-md5'][0] === '2NAu/QVqg211BbktgY5GkA=='
```
Comment (sindresorhus) [inline source/index.ts]: Can you move `'2NAu/QVqg211BbktgY5GkA=='` to a variable with a name that indicates what it is?

File: source/menu.ts [hunk 2]
```diff
@@ -247,6 +247,14 @@ Press Command/Ctrl+R in Caprine to see your changes.
 				sendAction('toggle-mute-notifications');
 			}
 		},
+		{
+			label: 'Mute Ringtone Call',
```
Comment (sindresorhus) [inline source/menu.ts]:
```suggestion
			label: 'Mute Call Ringtone',
```

File: source/index.ts [hunk 3]
```diff
@@ -233,6 +233,18 @@ function initRequestsFiltering(): void {
 			callback({cancel: config.get('block.deliveryReceipt' as any)});
 		}
 	});
+
+	session.defaultSession.webRequest.onHeadersReceived({
+		urls: ['*://static.xx.fbcdn.net/rsrc.php/*']
+	}, ({responseHeaders}, callback) => {
+		if (!config.get('callRingtoneMuted') || !responseHeaders) {
+			return callback({});
```
Comment (sindresorhus) [inline source/index.ts]:
```suggestion
		callback({});
		return;
```

**Comment by sindresorhus** [comment]
Have you manually tested that this actually works, by receiving a call and ensuring it doesn't make a sound?

**Comment by LuKks** [comment]
Yes. This is what I commented in the issue discussion some time ago:
> Due HTTP cache, for example, if the ringtone plays and now you disable it, the sound is already cached so you will need to restart the app.

I hope that is ok and I also tested it again, it works fine. By the way, really thank you for the corrections in the code.

**Pull request merged**